projects/models.nix

```
@@ -0,0 +1,128 @@
+{
+  lib,
+  pkgs,
+  sources,
+}:
+let
+  yants = import sources.yants { };
+
+  inherit (yants)
+    string
+    list
+    option
+    attrs
+    enum
+    either
+    struct
+    drv
+    path
+    restrict
+    ;
+
+  programType = struct "program" {
+    name = string;
```

```suggestion
    name = option string;
```

I don't see the need for a human-readable name *yet*, so we could leave that optional.